unit-of-work/src/main/java/com/iluwatar/unitofwork/IUnitOfWork.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2014-2017 Piyush Chaudhari
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.iluwatar.unitofwork;
+
+/**
+ * @param <T> Any generic entity
+ */
+public interface IUnitOfWork<T> {
+  String INSERT = "INSERT";
+  String DELETE = "DELETE";
+  String MODIFY = "MODIFY";
+
+  void registerNew(T entity);
```

Comment:
Suggestion: Javadocs on methods explaining that any modification operation occuring on UnitOfWork is only going to be performed on commit.

Reply:
@npathai : Make sense to have java doc on this register method to clarify it will be committed in commit method.
Thanks for pointing out.